### DISCUSSION TITLE
How to use `lazygit` to get the "branch-merged-into-main" curly line representation as that shown on `gitgraph` in `vscode`?

### DISCUSSION THREAD

**Discussion by hopezh** [jesseduffield/lazygit#2752] [Q&A] [answered]
# What I did: 
(Noted: What's shown on the top half of the image is `neovim` and `lazygit`, and the bottom half is `gitgraph` in `vscode`.)

1. I have the following test repo with a branch `lib`:
<img width="1267" alt="Screenshot 2023-07-01 at 12 15 41 PM" src="https://github.com/jesseduffield/lazygit/assets/23435810/eb2fd07d-66d2-451d-8fa9-1e94cac48929">

2. I changed the current branch to `main`,  and I used the `M` keybinding in `lazygit` to merge the seleted `lib` branch into `main`. 
<img width="1267" alt="Screenshot 2023-07-01 at 12 16 36 PM" src="https://github.com/jesseduffield/lazygit/assets/23435810/ad508087-74d1-49c2-bc79-e72817d044d3">

3. Note that in the `gitgraph` in `vscode` this merge is shown as a straight line. 
<img width="1267" alt="Screenshot 2023-07-01 at 12 16 51 PM" src="https://github.com/jesseduffield/lazygit/assets/23435810/592fabe3-66e6-4f23-ab87-81d0bc67020c">

# What I'm expecting:
4. Doing the same in `gitgraph` in `vscode`, we right-click the `lib` branch and select `Merge into current branch...`
<img width="1273" alt="Screenshot 2023-07-01 at 12 54 09 PM" src="https://github.com/jesseduffield/lazygit/assets/23435810/754df154-94f8-4623-b04c-042c35e936b9">

5. Click `Yes, merge` for the prompt:
<img width="1273" alt="Screenshot 2023-07-01 at 12 54 17 PM" src="https://github.com/jesseduffield/lazygit/assets/23435810/1d88e8b3-3f31-47b5-94e8-6f66da68b28b">

6. We will see that in the `gitgraph`, this merging action makes `lib` shown as a curly branching line merged into the straight line on the left. Similar graphic representation is also shown in `lazygit`
<img width="1273" alt="Screenshot 2023-07-01 at 12 54 28 PM" src="https://github.com/jesseduffield/lazygit/assets/23435810/ef6233e0-eca6-4dbd-a991-179f723fed5b">

# My question:
What is the command/keybinding in `lazygit` that can achieve the same graphical representations as shown in steps 4 to 6 above, i.e., a curly line, representing the branch being merged, leading back to the straight line on the left, representing the `main` branch?
<img width="1267" alt="Screenshot 2023-07-01 at 12 40 05 PM" src="https://github.com/jesseduffield/lazygit/assets/23435810/4a5d4625-7cd6-42db-bea9-e019048f9575">

**Comment by stefanhaller** [comment, marked as answer]
It looks like gitgraph is including the `--no-ff` option in its merge command, and lazygit doesn't. I like lazygit's approach better, because users who want this behavior can configure it on the git level by saying `git config --global merge.ff false`.

If you don't want to set it globally for all git clients, you can instead configure it for lazygit only by putting 
```
git:
  merging:
    args: '--no-ff'
```
into your lazygit config file; see https://github.com/jesseduffield/lazygit/blob/master/docs/Config.md#default.

**Reply by hopezh** [reply]
Thanks, @stefanhaller 

Yes, as you pointed out, adding `--no-ff` to `args` in the `config.yaml` of `lazygit` does the trick. 

The `git` cmd is : 
`git merge [branch-name] --no-ff -m "merge [branch-name]"`. 

I also add the `-m` option in `args` of lazygit's config, too. However, I'm not prompted to input a commit message while merging the branch. Do you know how to do this?

**Reply by stefanhaller** [reply]
There's no way to do that with lazygit. On the command line you would do that by passing `--edit`, but this has no effect in lazygit because the git command is not attached to a tty in this case.

**Reply by hopezh** [reply]
Noted with thanks. I think adding `--no-ff` to `args` in the `config.yml` is good enough.
